Dockerfile

```diff
@@ -0,0 +1,16 @@
+FROM node:8.1.2-alpine
+
+ENV HOME=/usr/windbreaker
+
+RUN mkdir -p $HOME
+
+# cache node_modules in a tmp dir to speed up builds
+ADD package.json /tmp/package.json
+RUN cd /tmp && npm install
+
+RUN mkdir $HOME/node_modules
+RUN cp -a /tmp/node_modules $HOME
```

Comment:
This part is cool little trick. We add the `package.json` to a temp dir and install modules in that dir. Later on we copy the files over our working directory. The next time the build runs, docker will diff the temp dir against what was cached to determine if a fresh install is needed. The only time a change will be detected is if the `package.json` is modified.